File: computer_science/algorithms/binary_search/binary-search.py


def binary_search(elements, search_data):
    down = 0
    up = len(elements) - 1

    while (down <= up):
        quite = round((down+up)/2)
        kick = elements[quite]

        if kick == search_data:
            return quite
        elif kick > search_data:
            up = quite - 1
        else:
            down = quite+1

File: computer_science/algorithms/binary_search/binary-search.rs
fn binary_search(elements: &[i32], search_data: i32) -> Option<usize> {
    let mut down = 0;
    let mut up = elements.len() - 1;
    while down <= up {
        let mid = (down + up) / 2;
        let mid_value = elements[mid];

        if mid_value == search_data {
            return Some(mid);
        } else if mid_value < search_data {
            down = mid + 1;
        } else {
            up = mid - 1;
        }
    }
    None
}
